ha/core/controllers/pcs/node_controller.py

```diff
@@ -280,31 +301,30 @@ def stop(self, node_id: str, timeout: int= -1, **op_kwargs) -> dict:
+        # Get the node_name (pvtfqdn) fron nodeid and raise exception if node_id is not valid
         node_name = self._get_node_name(node_id=node_id)
-        # TODO: check_cluster - boolean.It checks whether cluster is going to be offline if node with node_id is stopped.
-        timeout = const.NODE_STOP_TIMEOUT if timeout < 0 else timeout
-        # TODO: Get the node_status from system health status
-        node_status = self.nodes_status([node_name]).get(node_name)
-        if node_status == NODE_STATUSES.CLUSTER_OFFLINE.value:
-            Log.info(f"For stop {node_name}, Node already in offline state.")
-            status = f"Node {node_name} is already in offline state."
-        elif node_status == NODE_STATUSES.POWEROFF.value:
-            raise ClusterManagerError(f"Failed to stop {node_name}. Node is in {node_status}.")
-        else:
-            if self.heal_resource(node_name):
-                time.sleep(const.BASE_WAIT_TIME)
-            try:
-                Log.info(f"Please Wait, trying to stop node: {node_name}")
-                # TODO: Use PCS_STOP_NODE from const.py with timeout value
-                self._execute.run_cmd(f"pcs cluster stop {node_name}")
-                Log.info(f"Executed node stop for {node_name}, Waiting to stop resource")
-                time.sleep(const.BASE_WAIT_TIME)
-                status = f"Stop for {node_name} is in progress, waiting to stop resource"
-                # TODO: Update node_status in system_health
-            except Exception as e:
+        try:
+            stop_status = json.loads(super().stop(node_id, **op_kwargs))
+            if stop_status != None:
+                if stop_status["status"] == const.STATUSES.SUCCEEDED.value or stop_status["status"] == const.STATUSES.FAILED.value:
+                    return stop_status
+            Log.info(f"Please Wait, trying to stop node: {node_name}")
+            # TODO: Use PCS_STOP_NODE from const.py with timeout value
+            self._execute.run_cmd(f"pcs cluster stop {node_name} --force")
+            Log.info(f"Executed node stop for {node_name}, Waiting to stop resource")
+            time.sleep(const.BASE_WAIT_TIME)
+            status = f"Stop for {node_name} is in progress, waiting to stop resource"
+
+            # Update node health
+            # TODO : Health event update to be removed once fault_tolerance branch is merged
+            initial_event = self._system_health.get_health_event_template(nodeid=node_id, event_type=HEALTH_EVENTS.FAULT.value)
+            Log.debug(f"Node health : {initial_event} updated for node {node_name}")
+            health_event = HealthEvent.dict_to_object(initial_event)
+            self._system_health.process_event(health_event)
+            return {"status": const.STATUSES.SUCCEEDED.value, "output": status, "error": ""}
+
+        except Exception as e:
                 raise ClusterManagerError(f"Failed to stop {node_name}, Error: {e}")
```

Comment:
{node_name} use input node_id here as well.

Reply:
Changed to node_id.